receiver/k8sclusterreceiver/internal/collection/cronjobs.go

```diff
@@ -67,9 +88,28 @@ func getResourceForCronJob(cj *batchv1.CronJob) *resourcepb.Resource {
 	}
 }
 
+func getResourceForCronJobBeta(cj *batchv1beta1.CronJob) *resourcepb.Resource {
```

Comment:
Might not be worth it bc of generics, but could we prevent the duplicated logic between `getResourceForCronJob` and `getResourceForCronJobBeta` if instead of taking the version as the parameter we passed in the UID, Name, Namespace, and ClusterName?